sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -409,6 +409,18 @@ public void captureSession(final @NotNull Session session, final @Nullable Objec
         }
       }
     }
+    final List<Span> unfinishedSpans = new ArrayList<>();
+    for (Span span : transaction.getSpans()) {
+      if (!span.isFinished()) {
+        unfinishedSpans.add(span);
+      }
+    }
+    if (options.getLogger().isEnabled(SentryLevel.WARNING) && !unfinishedSpans.isEmpty()) {
```

Comment:
```suggestion
    if (!unfinishedSpans.isEmpty()) {
```

Reply:
`options.getLogger().log` already checks if its enabled internally, `!unfinishedSpans.isEmpty()` isn't a heavy call I believe, it'd be fine logging directly

Reply:
👍